static/app/components/onboarding/gettingStartedDoc/utils/useLoadGettingStarted.tsx

```diff
@@ -6,49 +6,73 @@ import {
   feedbackOnboardingPlatforms,
   replayPlatforms,
 } from 'sentry/data/platformCategories';
-import type {PlatformKey} from 'sentry/types/project';
+import type {Organization, PlatformIntegration, Project} from 'sentry/types';
+import {getPlatformPath} from 'sentry/utils/gettingStartedDocs/getPlatformPath';
+import {useProjectKeys} from 'sentry/utils/useProjectKeys';
 
-export default function useLoadGettingStarted({
-  platformId,
-  platformPath,
-  productType,
-}: {
-  platformId: PlatformKey;
-  platformPath: string;
+type Props = {
+  orgSlug: Organization['slug'];
+  platform: PlatformIntegration;
   productType?: 'feedback' | 'replay';
-}) {
-  const [module, setModule] = useState<
-    | null
-    | {
-        default: Docs<any>;
-      }
-    | 'none'
-  >(null);
+  projSlug?: Project['slug'];
+};
+
+export function useLoadGettingStarted({
+  platform,
+  productType,
+  orgSlug,
+  projSlug,
+}: Props): {
+  cdn: string | undefined;
+  docs: Docs<any> | null;
+  dsn: string | undefined;
+  isError: boolean;
+  isLoading: boolean;
+  refetch: () => void;
+} {
+  const [module, setModule] = useState<undefined | 'none' | {default: Docs<any>}>(
+    undefined
+  );
+
+  const projectKeys = useProjectKeys({orgSlug, projSlug});
+
+  const platformPath = getPlatformPath(platform);
 
   useEffect(() => {
     async function getGettingStartedDoc() {
       if (
-        (productType === 'replay' && !replayPlatforms.includes(platformId)) ||
-        (productType === 'feedback' && !feedbackOnboardingPlatforms.includes(platformId))
+        (productType === 'replay' && !replayPlatforms.includes(platform.id)) ||
+        (productType === 'feedback' && !feedbackOnboardingPlatforms.includes(platform.id))
       ) {
         setModule('none');
         return;
       }
+
       try {
         const mod = await import(
           /* webpackExclude: /.spec/ */
           `sentry/gettingStartedDocs/${platformPath}`
         );
         setModule(mod);
       } catch (err) {
+        setModule(undefined);
         Sentry.captureException(err);
       }
     }
+
     getGettingStartedDoc();
+
     return () => {
-      setModule(null);
+      setModule(undefined);
     };
-  }, [platformPath, platformId, productType]);
+  }, [platformPath, platform.id, productType]);
 
-  return module;
+  return {
+    refetch: projectKeys.refetch,
+    isLoading: projectKeys.isLoading || module === undefined,
+    isError: projectKeys.isError,
+    docs: module === 'none' ? null : module?.default ?? null,
```

Comment:
```suggestion
    docs: module?.default ?? null,
```
i think this should simplify the logic here

Reply:
That would be better, but we consider the module to be 'loading' when it is `undefined` and set it to `none` when certain conditions are met.

Reply:
if the module is set to `none`, then `module.default` doesn't exist right?